server/src/internalClusterTest/java/org/elasticsearch/action/admin/indices/create/AutoCreateSystemIndexIT.java

```diff
@@ -214,13 +235,45 @@ public void testAutoCreateSystemAliasViaComposableTemplate() throws Exception {
             ).get()
         );
 
-        String nonPrimaryIndex = INDEX_NAME + "-2";
+        String nonPrimaryIndex = indexName + "-2";
         CreateIndexRequest request = new CreateIndexRequest(nonPrimaryIndex);
         assertAcked(client().execute(AutoCreateAction.INSTANCE, request).get());
 
         assertTrue(indexExists(nonPrimaryIndex));
 
-        assertAliasesHidden(nonPrimaryIndex, Set.of(".test-index", ".test-index-composable-alias"));
+        return nonPrimaryIndex;
+    }
+
+    /**
+     * Check that a composable template does not create an alias for a system index
+     */
+    public void testAutoCreateSystemAliasViaComposableTemplate() throws Exception {
+        String nonPrimaryIndex = autoCreateSystemAliasViaComposableTemplate(INDEX_NAME);
+
+        assertAliasesHidden(nonPrimaryIndex, Set.of(INDEX_NAME), 1);
+
+        assertAcked(
```

Comment:
I'm pretty sure this is unnecessary as well despite not being explicitly in the `@After`, I ran it locally without this x10 and every thing, passed, and `:server:check` passed as well.

Reply:
I'm inclined to leave it, it does clean up the template explicitly and at the same time it tests we can clean up composable templates.